lib/features/rules/BpmnRules.js

```diff
@@ -1028,7 +1028,7 @@ function isOneTextAnnotation(source, target) {
 function canConnectAssociation(source, target) {
 
   // compensation boundary events are exception
-  if (isCompensationBoundary(source) && isForCompensation(target)) {
+  if (isCompensationBoundary(source) && is(target, 'bpmn:Activity') && !isEventSubProcess(target)) {
```

Comment:
This likely prevents connecting event sub processes to text annotations?

![capture l7Mo9k_optimized](https://github.com/bpmn-io/bpmn-js/assets/58601/07cae994-6e0c-49da-b6fd-d3e165d0609d)